interfaces/hotplug/udevadm.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+var udevadmBin = `/sbin/udevadm`
+var udevadmRe = regexp.MustCompile("^([A-Z]): (.*)$")
+
+// RunUdevadm enumerates all devices by parsing 'udevadm info -e' command output and reports them
+// via devices channel. The devices channel gets closed to indicate that all devices were processed.
+// Parsing errors are reported to parseErrors channel. Fatal errors encountered when starting udevadm
+// are reported by the error return value.
+func RunUdevadm(devices chan<- *HotplugDeviceInfo, parseErrors chan<- error) error {
+	cmd := exec.Command(udevadmBin, "info", "-e")
```

Comment:
I think this should be `exec.Command(filepath.Join(dirs.GlobalRootDir, udevadmBin),...`

Reply:
Thanks for pointing this out. I've actually reworked this a little bit to follow what we do for other commands (such as `unsquashfs`). I don't build the path anymore, command location relies on PATH and mocking is done via existing MockCommand helper (I got rid of my custom MockUdevadm function).